components/plurals/src/data.rs

```diff
@@ -91,6 +91,18 @@ impl RulesSelector {
                 .unwrap_or(PluralCategory::Other),
         }
     }
+
+    /// Returns an iterator over each [`PluralCategory`] for which this [`RulesSelector`]
+    /// has rules for.
+    ///
+    /// The category [`PluralCategory::Other`] is always included.
+    pub fn categories(&self) -> impl Iterator<Item = &'static PluralCategory> + '_ {
+        match &self {
+            Self::Conditions(conditions) => PluralCategory::all().filter(move |&category| {
+                category.eq(&PluralCategory::Other) || conditions.get(*category).is_some()
```

Comment:
`suggestion (nb)` - this is very nitpicky, but since I noticed, I may as well share - you're using `get` on `conditions` when you only really want to know if `conditions` contain a category. It might be tad cheaper assembly to add `contains(key) -> bool` to `PluralRuleList`.

Reply:
> follow-up

I think this is great feedback. It's a simple change, and probably will produce better code. 

I decided to call it `has_rules_for()` rather than `contains()` since the `PluralRuleList` does not contain a member for `Other` but there is always a rule for `Other`, since it is implicitly the fallback. 

This allows me to return `true` for the other case, removing the or-expression that checks for `Other` every time within the `filter` closure. 

I think this is a cleaner solution, even if slightly confusing, but this is not a public function, and is only used in this one place for this context.